jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-backend/src/main/java/org/jbpm/console/ng/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -159,5 +164,39 @@ public void removeComment(final String serverTemplateId, final String containerI
         return comparing(CaseMilestoneSummary::getStatus).thenComparing(request.getSortByAsc() ? comparatorByName: comparatorByName.reversed());
     }
 
+    @Override
+    public CaseActionsLists getCaseActionsLists(String containerId, String caseId, String userId) {
+        List<CaseActionSummary> availableActions = getCaseActionsByStatus(caseId, userId, "Ready");
+        availableActions.addAll(client.getAdHocFragments(containerId, caseId)
+                .stream()
+                .map(new CaseActionAdhocFragmentMapper())
+                .sorted(comparing(CaseActionSummary::getName)).collect(toList()));
+
+        CaseActionsLists caseActionsLists = new CaseActionsLists();
+        caseActionsLists.setAvailableActionList(availableActions);
+        caseActionsLists.setInprogressActionList(getCaseActionsByStatus(caseId, userId, "InProgress", "Reserved"));
```

Comment:
I also have the feeling that case statuses are going to be used at more than one place throughout case mgmt app. Wouldn't it be better to represent them using Enum for better type safety and protection from typos?